### ISSUE TITLE
Mistaken path change from "images" to "labels"

### ISSUE THREAD

**Issue by tetelevm** [ultralytics/yolov5#3719]
Hello! I create dir `/images/train/` on my GoogleDrive and put the train images into `/images/train/images/` and labels into `/images/train/labels/`. I expect the script to look for labels in `/images/train/labels`, but it looks at the `/labels/to_train/images/` (replaces the first word "images" instead of the last word). Isn't this a bug?

**Comment by glenn-jocher** [comment]
@tetelevm yes I see the same result in master. Code to reproduce:

```python
import os
from utils.datasets import img2label_paths

img2label_paths(['dataset/images/images/img1.jpg'])

# output ['dataset/labels/images/img1.txt']
```

**Comment by glenn-jocher** [comment]
Possible fix:

```python
def img2label_paths_fix(img_paths):
    # Define label paths as a function of image paths
    sa, sb = os.sep + 'images' + os.sep, os.sep + 'labels' + os.sep  # /images/, /labels/ substrings
    return ['txt'.join(sb.join(x.rsplit(sa, 1)).rsplit(x.split('.')[-1], 1)) for x in img_paths]
```

```python
import os
from utils.datasets import img2label_paths, img2label_paths2
print(img2label_paths(['dataset/images/images/img1.jpg']))
print(img2label_paths_fix(['dataset/images/images/img1.jpg']))
%timeit img2label_paths(['dataset/images/images/img1.jpg'] * 100)
%timeit img2label_paths_fix(['dataset/images/images/img1.jpg'] * 100)
['dataset/labels/images/img1.txt']
['dataset/images/labels/img1.txt']
57.9 µs ± 2.21 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
73.6 µs ± 3.73 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

**Comment by glenn-jocher** [comment]
Possible fix:

```python
import os

def img2label_paths(img_paths):
    # Define label paths as a function of image paths
    sa, sb = os.sep + 'images' + os.sep, os.sep + 'labels' + os.sep  # /images/, /labels/ substrings
    return ['txt'.join(x.replace(sa, sb, 1).rsplit(x.split('.')[-1], 1)) for x in img_paths]


def img2label_paths_fix(img_paths):
    # Define label paths as a function of image paths
    sa, sb = os.sep + 'images' + os.sep, os.sep + 'labels' + os.sep  # /images/, /labels/ substrings
    return ['txt'.join(sb.join(x.rsplit(sa, 1)).rsplit(x.split('.')[-1], 1)) for x in img_paths]


def img2label_paths_fix2(img_paths):
    # Define label paths as a function of image paths
    sa, sb = os.sep + 'images' + os.sep, os.sep + 'labels' + os.sep  # /images/, /labels/ substrings
    return [sb.join(x.rsplit(sa, 1)).rsplit('.')[0] + '.txt' for x in img_paths]

images = ['dataset/images/images/img1.jpg'] * 100

print(img2label_paths(images[0:1]))
print(img2label_paths_fix(images[0:1]))
print(img2label_paths_fix2(images[0:1]))

%timeit img2label_paths(images)
%timeit img2label_paths_fix(images)
%timeit img2label_paths_fix2(images)

## OUTPUT
['dataset/labels/images/img1.txt']
['dataset/images/labels/img1.txt']
['dataset/images/labels/img1.txt']

55.2 µs ± 519 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
70.7 µs ± 2.09 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
51.8 µs ± 336 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

It looks like we have a winner.

**Comment by glenn-jocher** [comment]
@tetelevm good news 😃! Your original issue may now be fixed ✅ in PR #3720. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
